osu.Game/Screens/Edit/Changes/QueueUpdateHitObject.cs

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Game.Rulesets.Objects;
+
+namespace osu.Game.Screens.Edit.Changes
+{
+    /// <summary>
+    /// Queues the update of a <see cref="HitObject"/> in an <see cref="EditorBeatmap"/> for undo/redo.
+    /// The order of the updates in the transaction does not matter, because the updates are aggregated and applied on the next frame.
+    /// </summary>
+    public class QueueUpdateHitObject : IRevertibleChange
```

Comment:
As stated [previously](https://github.com/ppy/osu/pull/30255#issuecomment-2416204810), `HitObject.Update()` doesn't really make sense as a command / "revertible change" or however you put it. It is always supposed to be part of some larger operation.

Reply:
Yeah it doesnt truly fit in the category of "revertible change". It was put here as sort of a hack to make sure the correct hit objects get updated on undo. I need some way to know which objects to update on undo/redo.

I think a better approach would be to store the set of hit objects to update in `Transaction` directly adjacent to the list of changes, and then either providing these on the `BeginChange()` call or capture them from whenever `EditorBeatmap.Update()` is called.

The alternative solution is to make it so you never have to call `EditorBeatmap.Update` at all. I think that sort of change is better left for later though. I have some ideas on how it could be done but they all have downsides:
- Have the change `Apply()` call the update. Would require all changes to have the `EditorBeatmap` and the relevant hit object.
- Have `EditorBeatmap` listen for submitted changes that target hit objects. Would need some interface to get the `Target` object from the change.
- Make everything in `HitObject` bindable with `HitObjectProperty`. Might add a lot of overhead.